### PR TITLE
Using ORAS digest & beta tags in Github action

### PR DESCRIPTION
**Pull request by ukrocks007**
Thank you for contributing to Retraced!

# Please add a summary of your change

# Does your change fix a particular issue?

Fixes #(issue)

### CODE REVIEW

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -107,7 +111,7 @@ jobs:
           file: ./deploy/Dockerfile-slim
           platforms: linux/amd64,linux/arm64
           push: true
-          tags: ${{ secrets.DOCKER_HUB_USERNAME }}/retraced:${{ needs.ci.outputs.PUBLISH_TAG }},${{ secrets.DOCKER_HUB_USERNAME }}/retraced:${{ steps.slug.outputs.SHA7 }},${{ secrets.DOCKER_HUB_USERNAME }}/retraced:${{ needs.ci.outputs.NPM_VERSION }}
+          tags: ${{ secrets.DOCKER_HUB_USERNAME }}/retraced${{ needs.ci.outputs.IMAGE_SUFFIX }}:${{ needs.ci.outputs.PUBLISH_TAG }},${{ secrets.DOCKER_HUB_USERNAME }}/retraced${{ needs.ci.outputs.IMAGE_SUFFIX }}:${{ steps.slug.outputs.SHA7 }},${{ secrets.DOCKER_HUB_USERNAME }}/retraced${{ needs.ci.outputs.IMAGE_SUFFIX }}:${{ needs.ci.outputs.NPM_VERSION }}
```
Comment (deepakprabhakara) [inline .github/workflows/main.yml:114]: Move `${{ secrets.DOCKER_HUB_USERNAME }}/retraced${{ needs.ci.outputs.IMAGE_SUFFIX }}` into a var

**Review by deepakprabhakara — CHANGES_REQUESTED** [review]

File: .github/workflows/main.yml [hunk 2]
```diff
@@ -220,23 +224,33 @@ jobs:
...
-          ../oras_install/oras push ghcr.io/${{github.repository_owner}}/retraced/sbom:docker-${{ needs.ci.outputs.NPM_VERSION }} ./sbom.*
-          cd ..
 
-      - name: Sign the sbom images
+      - name: Push SBOM reports to GitHub Container Registry & Sign the sbom images
         if: github.ref == 'refs/heads/release'
         run: |
-          cosign sign -y --key /tmp/cosign.key ghcr.io/${{github.repository_owner}}/retraced/sbom:service-${{ needs.ci.outputs.NPM_VERSION }}
-          cosign sign -y --key /tmp/cosign.key ghcr.io/${{github.repository_owner}}/retraced/sbom:docker-${{ needs.ci.outputs.NPM_VERSION }}
+          result=$(./oras_install/oras push ghcr.io/${{github.repository_owner}}/retraced/sbom${{ needs.ci.outputs.IMAGE_SUFFIX }}:service-${{ needs.ci.outputs.NPM_VERSION }} ./sbom.*)
```
Comment (deepakprabhakara) [inline .github/workflows/main.yml:239]: Shouldn't this just be `${{ github.repository }}`?

**Review by deepakprabhakara — CHANGES_REQUESTED** [review]

**Review by deepakprabhakara — APPROVED** [review]

File: .github/workflows/main.yml [hunk 3]
```diff
@@ -23,6 +23,7 @@ jobs:
     outputs:
       NPM_VERSION: ${{ steps.version.outputs.NPM_VERSION }}
       PUBLISH_TAG: ${{ steps.version.outputs.PUBLISH_TAG }}
+      IMAGE_SUFFIX: ${{ steps.version.outputs.IMAGE_SUFFIX }}
```
Comment (deepakprabhakara) [inline .github/workflows/main.yml:26]: Shouldn't IMAGE_PATH be added here?

**Review by deepakprabhakara — CHANGES_REQUESTED** [review]
IMAGE_PATH is missing from outputs

File: .github/workflows/main.yml (hunk 3 above)
Comment (ukrocks007) [reply .github/workflows/main.yml:26]: I tried that way, but it came as blank in the `Build & Push` step.
So I shifted the `IMAGE_PATH` creation to the `Slug` step & it worked
So we don't need to add `IMAGE_Path` here.

**Comment by ukrocks007** [comment]
> IMAGE_PATH is missing from outputs

not required as it is created in the Build step only.

**Review by deepakprabhakara — APPROVED** [review]

**Pull request merged**